agent/app/agent_capability.go

```diff
@@ -458,7 +458,11 @@ func (agent *ecsAgent) appendServiceConnectCapabilities(capabilities []*ecs.Attr
 		}
 	}
 
-	return appendNameOnlyAttribute(capabilities, attributePrefix+capabilityServiceConnect)
+	supportedAppnetInterfaceVerToCapabilities := agent.serviceconnectManager.GetSupportedAppnetInterfaceVerToCapabilities()
```

Comment:
Ditto - the knowledge of "which appnet version are we actually using" should come from loader as it's a runtime decision.